scipy/spatial/distance.py

```diff
@@ -1086,43 +1100,66 @@ def sokalsneath(u, v):
     return float(2.0 * (ntf + nft)) / denom
 
 
-# Registry of "simple" distance metrics' pdist and cdist implementations,
-# meaning the ones that accept one dtype and have no additional arguments.
-_SIMPLE_CDIST = {}
-_SIMPLE_PDIST = {}
-
-for wrap_name, names, typ in [
-    ("bray_curtis", ['braycurtis'], "double"),
-    ("canberra", ['canberra'], "double"),
-    ("chebyshev", ['chebychev', 'chebyshev', 'cheby', 'cheb', 'ch'], "double"),
-    ("city_block", ["cityblock", "cblock", "cb", "c"], "double"),
-    ("euclidean", ["euclidean", "euclid", "eu", "e"], "double"),
-    ("sqeuclidean", ["sqeuclidean", "sqe", "sqeuclid"], "double"),
-    ("dice", ["dice"], "bool"),
-    ("kulsinski", ["kulsinski"], "bool"),
-    ("rogerstanimoto", ["rogerstanimoto"], "bool"),
-    ("russellrao", ["russellrao"], "bool"),
-    ("sokalmichener", ["sokalmichener"], "bool"),
-    ("sokalsneath", ["sokalsneath"], "bool"),
-    ("yule", ["yule"], "bool"),
-]:
-    converter = {"bool": _convert_to_bool,
-                 "double": _convert_to_double}[typ]
-    fn_name = {"bool": "%s_bool_wrap",
-               "double": "%s_wrap"}[typ] % wrap_name
-    cdist_fn = getattr(_distance_wrap, "cdist_%s" % fn_name)
-    pdist_fn = getattr(_distance_wrap, "pdist_%s" % fn_name)
-    for name in names:
-        _SIMPLE_CDIST[name] = converter, cdist_fn
-        _SIMPLE_PDIST[name] = converter, pdist_fn
-
-_METRICS_NAMES = ['braycurtis', 'canberra', 'chebyshev', 'cityblock',
-                  'correlation', 'cosine', 'dice', 'euclidean', 'hamming',
-                  'jaccard', 'kulsinski', 'mahalanobis', 'matching',
-                  'minkowski', 'rogerstanimoto', 'russellrao', 'seuclidean',
-                  'sokalmichener', 'sokalsneath', 'sqeuclidean', 'yule', 'wminkowski']
-
-_TEST_METRICS = {'test_' + name: eval(name) for name in _METRICS_NAMES}
+_convert_to_double = partial(_convert_to_type, out_type=np.double)
+_convert_to_bool = partial(_convert_to_type, out_type=bool)
+
+# adding python-only wrappers to _distance_wrap module
+_distance_wrap.pdist_correlation_double_wrap = _correlation_pdist_wrap
+_distance_wrap.cdist_correlation_double_wrap = _correlation_cdist_wrap
+
+# Registry of implemented metrics:
+# Dictionary with the following structure:
+# {
+#  metric_name : MetricInfo(aka, types=[double])
+# }
+#
+# `metric_name` must be equal to python metric name
+#
+# MetricInfo is a named tuple with fields:
+#  'aka' : [list of aliases],
+#
+#  'types': [list of supported types],  # X (pdist) and XA (cdist) are used to
+#                                       # choose the type. if there is no match
+#                                       # the first type is used. Default double
+#
+#}
+MetricInfo = namedtuple("MetricInfo", 'aka types')
+MetricInfo.__new__.__defaults__ = tuple([['double']])
```

Comment:
Nice, didn't know about `namedtuple` defaults!

Why not `(['double],)`? Do you find writing out `tuple` more readable?